modules/docs/arrow-docs/docs/docs/typeclasses/traverse/README.md

```diff
@@ -9,8 +9,259 @@ permalink: /docs/typeclasses/traverse/
 {:.intermediate}
 intermediate
 
-TODO. Meanwhile you can find a short description in the [intro to typeclasses]({{ '/docs/typeclasses/intro/' | relative_url }}).
+`Traverse`, also known as `Traversable` is a Typeclass used to perform traversals over an structure with an effect.
 
+We will see some of the use cases for Traverse. Let's start by looking for an example where the side effects are modeled as data types. Side effects in functional programming are changes outside of the scope of the function, for example, performing some IO operation, modifying global variables, etc...
+
+In Kotlin with Arrow, these aforementioned data types can be modeled as `Option` for missing values, `Either` and `Validated` for things that could either provide a right result or give an error, and `IO`, `Async` for asynchronous computations.
+
+
+Let's show some examples that will need the following imports.
+
+```kotlin:ank:silent
+import arrow.Kind
+import arrow.core.*
+import arrow.data.*
+import arrow.effects.*
+import arrow.effects.deferredk.applicative.applicative
+import arrow.instances.either.applicative.applicative
+import arrow.instances.either.applicativeError.applicativeError
+import java.lang.NumberFormatException
+import arrow.instances.list.applicative.*
+import arrow.instances.list.foldable.sequence_
+import arrow.instances.list.foldable.traverse_
+import arrow.instances.list.traverse.sequence
+import arrow.instances.list.traverse.traverse
+import arrow.instances.validated.applicative.applicative
+import arrow.instances.nonemptylist.semigroup.semigroup
+import arrow.instances.option.applicative.applicative
+import kotlinx.coroutines.*
+```
+
+Next, we will define some data classes and functions to show case different examples of traverse below.
+
+```kotlin:ank:silent
+sealed class SecurityError {
+    data class RuntimeSecurityError(val cause: String) : SecurityError()
+}
+
+interface Credential
+data class Profile(val id: String)
+data class User(val id: String, val name: String)
+```
+
+`parseInt`: is a function that will try to convert an `String` parameter `s` to an `Int`, it if succeeds it will return the number inside the `Some`. If it fails, it will return `None`.
```

Comment:
```suggestion
`parseInt`: is a function that will try to convert an `String` parameter `s` to an `Int`, it if succeeds it will return the number inside `Some`. If it fails, it will return `None`.
```

Reply:
changed